core/payment/src/api/allocations.rs

```diff
@@ -46,44 +112,147 @@ async fn create_allocation(
+                        }
+                    };
+                    log::debug!(
+                        "Selected network with default token {driver}-{network}-{default_token}"
+                    );
+                    format!("{}-{}-{}", driver, network, default_token)
+                }
+            }
+        }
+        None => {
+            log::debug!("No paymentPlatform entry found, using {DEFAULT_PAYMENT_PLATFORM}");
+            DEFAULT_PAYMENT_PLATFORM.to_string()
+        }
     };
 
     let address = allocation
         .address
         .clone()
         .unwrap_or_else(|| node_id.to_string());
 
-    // If the request contains information about the payment platform, initialize the account
-    // by setting the `send` field to `true`, as it is implied by the intent behing allocation of funds.
-    if let Some(platform) = &allocation.payment_platform {
-        // payment_platform is of the form driver-network-token
-        // eg. erc20-rinkeby-tglm
-        let [driver, network, _token]: [&str; 3] =
-            match platform.split('-').collect::<Vec<_>>().try_into() {
-                Ok(arr) => arr,
-                Err(_e) => {
-                    return response::bad_request(
-                        &"paymentPlatform must be of the form driver-network-token",
-                    )
-                }
-            };
-
-        let acc = Account {
-            driver: driver.to_owned(),
-            address: address.clone(),
-            network: Some(network.to_owned()),
-            token: None,
-            send: true,
-            receive: false,
-        };
-
-        if let Err(e) = init_account(acc).await {
-            log::error!("Error initializing account: {:?}", e);
-            return response::server_error(&e);
+    // payment_platform is of the form driver-network-token
+    // eg. erc20-rinkeby-tglm
+    let [driver, network, token]: [&str; 3] = match payment_platform
+        .split('-')
+        .collect::<Vec<_>>()
+        .try_into()
+    {
+        Ok(arr) => arr,
+        Err(_e) => {
+            let err_msg = format!("paymentPlatform must be of the form driver-network-token instead of {payment_platform}");
+            log::error!("{}", err_msg);
+            return response::bad_request(&err_msg);
         }
+    };
+
+    if let Err(err) = validate_network(network) {
+        let err_msg = format!("Validate network failed (2): {err}");
+        log::error!("{}", err_msg);
+        return response::bad_request(&err_msg);
+    }
+    if let Err(err) = validate_driver(network, driver) {
+        let err_msg = format!("Validate driver failed (2): {err}");
+        log::error!("{}", err_msg);
+        return response::bad_request(&err_msg);
+    }
+    if let Err(err) = validate_token(network, driver, token) {
+        let err_msg = format!("Validate token failed (2): {err}");
+        log::error!("{}", err_msg);
+        return response::bad_request(&err_msg);
+    }
```

Comment:
From api you get these values already split. Why do you first build this string to dispatch it later?
You could avoid all these duplicated checks here